LocalizationEditorTests/LoalizationProviderUpdatingTests.swift

```diff
@@ -75,21 +129,29 @@ class LoalizationProviderUpdatingTests: XCTestCase {
                 // nothing was deleted but a missing value might have been added
                 XCTAssert(baseKeys.count <= updatedKeys.count)
 
-                if let changesForLanguage = changes[base[i].localizations[j].language] {
-                    let existingKeys = updated[i].localizations[j].translations.filter { (string: LocalizationString) -> Bool in
-                        return changesForLanguage.map({$0.key}).contains(string.key)
+                // The current method only counts changed keys/values, messages are not counted as change. Therefore this flag prevents test from failing whre only a message was changed. The equality of messages are checked later.
+                if !onlyMessagesChanged {
+                    if let changesForLanguage = changes[base[i].localizations[j].language] {
+                        let existingKeys = updated[i].localizations[j].translations.filter { (string: LocalizationString) -> Bool in
+                            return changesForLanguage.map({$0.key}).contains(string.key)
+                        }
+                        XCTAssertEqual(changesForLanguage.count, existingKeys.count)
                     }
-                    XCTAssertEqual(changesForLanguage.count, existingKeys.count)
                 }
 
                 for key in baseKeys {
                     let originalValue = base[i].localizations[j].translations.first(where: { $0.key == key })?.value
                     let updatedValue = updated[i].localizations[j].translations.first(where: { $0.key == key })?.value
 
                     XCTAssertFalse(originalValue == nil)
+                    
+                    let originalMessage = base[i].localizations[j].translations.first(where: { $0.key == key })?.message
+                    let updatedMessage = updated[i].localizations[j].translations.first(where: { $0.key == key })?.message
+                    
 
                     if let lang = changes[base[i].localizations[j].language], let newValue = lang[key] {
                         XCTAssertEqual(updatedValue, newValue)
+                        XCTAssertEqual(originalMessage, updatedMessage)
```

Comment:
Message equality should always be tested, it would catch the bug from `LocalizationProvider`